planner/planner_APIs.go

```diff
@@ -440,6 +470,67 @@ func (planner *MyPlanner) getPlanningApi(ctx *gin.Context) {
 	utils.LogErrorWithLevel(planner.ResultHTMLTemplate.Execute(ctx.Writer, planningResp), utils.LogError)
 }
 
+func (planner *MyPlanner) getPlanDetails(c *gin.Context) {
+
+	var id string = c.Param("id")
+	iowrappers.Logger.Debugf("GET Route /plans/%s", id)
+
+	var planRecordRedisKey = strings.Join([]string{iowrappers.TravelPlanRedisCacheKeyPrefix, id}, ":")
+	cachePlanSolution, cacheErr := planner.RedisClient.GetCachePlan(c, planRecordRedisKey)
+	if cacheErr != nil {
+		iowrappers.Logger.Debugf("Key: %s does not exist!\n", planRecordRedisKey)
+		c.String(http.StatusBadRequest, cacheErr.Error())
+		return
+	}
+
+	// FIXME: [Rui] check why placeIDs are not stored in order in cached slot solution
+	const fixedPlaceKeyPrefix = "place_details:place_ID:"
+	var placeKey string
+	var destination string = "Dream Place"
+	if cachePlanSolution.Destination != (POI.Location{}) {
+		destination = strings.Title(cachePlanSolution.Destination.City) + ", " + strings.ToUpper(cachePlanSolution.Destination.Country)
+	}
+	var tripResp = TripDetailResp{
+		PlaceDetails:      make([]PlaceDetailsResp, 0),
+		ShownActive:       make([]bool, 0),
+		TravelDestination: destination,
+	}
+	for idx, placeId := range cachePlanSolution.PlaceIDs {
+		if idx >= 3 {
```

Comment:
What does this `>=3` mean?

Reply:
I initially added this condition to avoid excessive API requests for each plan. It seems the rest of codes are stable now, and this condition is redundant, so I have removed it.